Form submit example and req object args doc

layik opened the pull request:
It is not easy to be consistent with style and flow of the docs but happy to tweak PR to be acceptable.

layik committed:
Form submit example and req object args doc

layik committed:
03-routing-and-input subit form PR: minor, matching end points and valid html form

trestletech commented on 03-routing-and-input.Rmd:
```diff
@@ -323,4 +324,28 @@ You can print out the names of all of the properties attached to the request by
+<h2>Basic login form</h2>
+
+<pre>
+    <form action="/login">
```
I know the default is POST, but it might be helpful to new folks to specify the `method` tag.

layik replied:
Done.

trestletech commented on 03-routing-and-input.Rmd:
```diff
@@ -323,4 +324,28 @@ You can print out the names of all of the properties attached to the request by
+<pre>
+    <form action="/login">
+      <input type="text" placeholder="Enter Username" name="uname" required>
+      <input type="password" placeholder="Enter Password" name="psw" required>
```
Let's use a different example than username/password since that brings up a number of security considerations. Also some browsers will now render scary warnings if you try to enter a password on non-HTTPS.

Maybe we could do email address and opt-in or something trivial: favorite color text input and a checkbox for something.

layik replied:
Done.

trestletech commented on files/apis/03-06-args.R:
```diff
@@ -0,0 +1,4 @@
+#' @post /login
+function(req){
+  cat(req$args$uname, req$args$psw)
+}
```
This API could go ahead and host the HTML file, too.

layik replied:
Done.

layik replied:
Be it using the `@html` tag!

trestletech commented on 03-routing-and-input.Rmd:
```diff
@@ -238,6 +238,7 @@ Name | Example | Description
 `SERVER_NAME` | `"127.0.0.1"` | The host portion of the incoming request. You may favor `HTTP_HOST`, if available.
 `SERVER_PORT` | `"8000"` | The target port for the request
 `HTTP_*` | `"HTTP_USER_AGENT"` | Entries for all of the HTTP headers sent with this request.
+`args` | `"a=123"` | A named vector of the `QUERY_STRING`
```
I worry that the example here may be misleading since it's presented as a string. Perhaps we could use the syntax we use with `cookies` above. And pardon my rusty memory, but is the args value a proper list? Or is it just a named vector?

layik replied:
Was contradicting myself, list indeed.

trestletech committed:
Minor wording nits.

trestletech reviewed (approved):
Thanks for contributing! I think this example will be a nice addition.

I commented with a few questions/suggestions and committed a few wording nits. Let me know what you think about those changes.

layik committed:
Change from uname/pass to more generic form with checkbox, tweak pull

layik commented:
Hi @trestletech (Jeff). Slight delay and seems a little hiccup with branches. I think I have addressed the comments in the latest commit. Please have a look.

trestletech commented on files/apis/03-06-args.R:
```diff
@@ -0,0 +1,22 @@
+  <form action="http://localhost:8000/submit" method="post">
+      <input type="text" placeholder="Email" name="email" required>
+      <input type="checkbox" name="bike" value="Bike"> I have a bike
+      <input type="text" placeholder="Bike colour" name="color">
```
I'm fine with `colour` or `color`, but let's make the Placeholder/name consistent.

layik replied:
Done

trestletech commented on files/apis/03-06-args.R:
```diff
@@ -0,0 +1,22 @@
+#' @post /submit
+#' get /submit
```
Can we kill this line? We don't need @get and right now it's missing the `@` to be a real annotation anyways.

layik replied:
Done

trestletech commented on 03-routing-and-input.Rmd:
````diff
@@ -323,4 +324,40 @@ You can print out the names of all of the properties attached to the request by
+```html
+<h2>Basic submit form</h2>
+
+<pre>
````
Why the `<pre>`?

layik replied:
Done.

trestletech commented on 03-routing-and-input.Rmd:
````diff
@@ -323,4 +324,40 @@ You can print out the names of all of the properties attached to the request by
+</pre>
+```
+
+and  a `/submit` end-point as below:
````
`endpoint`, please

layik replied:
Done.

trestletech commented on 03-routing-and-input.Rmd:
```diff
@@ -323,4 +324,40 @@ You can print out the names of all of the properties attached to the request by
+
+The `name` in the inputs correspond to the `names` of the `req$args` list passed to the Plumber endpoint.
+
+To run/test the example above, just run the file in `files/apis/03-06-args.R` of the repo using `plumber` as follows:
```
I usually don't presume that people have the repo checked out. Instead of referencing the file address, let's just presume that they can figure out how to run the API that will be printed above.

layik replied:
Done.

trestletech commented on 03-routing-and-input.Rmd:
````diff
@@ -323,4 +324,40 @@ You can print out the names of all of the properties attached to the request by
+
+```{r, eval=FALSE}
+p = plumber::plumb("files/apis/03-06-args.R")
+p$run(port = 8000)
````
I think it is good to keep this point, though. They will need to run on port 8000 if the HTML is going to work. So let's keep the clarification that it presumes that they're going to run on 8000.

layik replied:
Agree.

trestletech reviewed (changes requested):
Thanks for another round! A couple of points to address, but I think we're close.

layik committed:
Addressing comments on PR

layik commented:
Did not build. Hope builds OK.

layik commented:
Hi @trestletech, we/you must be aware. 

I just noticed that this branch was actually "forked" from master which includes the other PR which, I know I should have dealt with before this one, so be aware. Happy to do some "rebase" if I have to be fore the final approval. I do not think GH is as smart as keeping those commits out!

layik commented:
I misunderstood what was in the branch. As it the other PR has not been accepted, naturally it won't be part of this PR to the "main" repo. I think this PR can be accepted unless there are more tweaks to be done.

layik commented:
```
413f50e (HEAD -> args, origin/args) Addressing comments on PR
41e228a Change from uname/pass to more generic form with checkbox, tweak pull
35627f3 Minor wording nits.
f46d1f4 03-routing-and-input subit form PR: minor, matching end points and valid htmlform
a4c155d Form submit example and req objectargs doc
02114b6 Merge pull request #13 from thalesmello/patch-1
a2226f6 Fix typo in HTML link
8474a10 Merge pull request #12 from layik/patch-1
3606106 Nits
f0d497f Clarification regarding default/index.html
c608cf9 m0ar pandoc?
3980581 Update pandoc
0930bb0 Minor wording fixes
155ebeb Newline spacing```

layik commented:
ping @trestletech.
